src/classes/Bot.ts

```diff
@@ -772,6 +777,17 @@ export default class Bot {
                             return callback(null);
                         });
                     },
+                    (callback): void => {
+                        if (this.options.discordApiToken) {
+                            log.debug(`Initializing Discord bot...`);
+                            this.discordBot = new DiscordBot(this.options, this);
+                            void this.discordBot.start();
+                        } else {
+                            log.debug('Discord api key is not set, ignoring.');
+                        }
+                        /* eslint-disable-next-line @typescript-eslint/no-unsafe-return, @typescript-eslint/no-unsafe-call */
+                        return callback(null);
+                    },
```

Comment:
No need to use try-catch on here (Bot.ts).
Just
```ts
                    (callback): void => {
                        if (this.options.discordApiToken) {
                            log.debug(`Initializing Discord bot...`);
                            this.discordBot = new DiscordBot(this.options, this);
                            void this.discordBot.start()
                                .then(() => {
                                    /* eslint-disable-next-line @typescript-eslint/no-unsafe-return, @typescript-eslint/no-unsafe-call */
                                    return callback(null);
                             })
                               .catch(err => {
                                   /* eslint-disable-next-line @typescript-eslint/no-unsafe-return, @typescript-eslint/no-unsafe-call */
                                   return callback(err);
                             });
                        } else {
                            log.debug('Discord api key is not set, ignoring.');
                            /* eslint-disable-next-line @typescript-eslint/no-unsafe-return, @typescript-eslint/no-unsafe-call */
                            return callback(null);
                        }
                    },
```

Reply:
Now I see the logic of implementing it as a promise.

But do we really need to crash the bot on startup? Error on login is not always user's mistake, Discord can simply be unavailable. Maybe in this case will be better to send to Steam chat something like "Error while reaching Discord, check logs or use Steam chat"?

Reply:
If we just send that, then they might just restart the bot again and again until it works, unless the bot automatically retries if it's a server-side problem (status code >= 500).

Reply:
Then I offer to crash on 401 and continue with Steam chat in other cases. Possible message: "if https://discordstat.us doesn't indicate any problems right now, you can try to restart. If restarting didn't fix the problem - please ask for help".

Reply:
Alright, that looks good. 👍

Reply:
In fact, that's not 401, but 4004 (Gateway Close Event Code). But it doesn't really change the logic.

Reply:
I see. Thank you very much.